### ISSUE TITLE
'git.latest' raises a KeyError when `git` is not installed.

### ISSUE THREAD

**Issue by jhgg** [saltstack/salt#26536]
```python
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1161, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 523, in highstate
        whitelist=kwargs.get('whitelist')
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2968, in call_highstate
        return self.state.call_high(high)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2067, in call_high
        ret = dict(list(disabled.items()) + list(self.call_chunks(chunks).items()))
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1623, in call_chunks
        running = self.call_chunk(low, running, chunks)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1858, in call_chunk
        running = self.call_chunk(chunk, running, chunks)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1769, in call_chunk
        self._mod_init(low)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 612, in _mod_init
        self.states['{0}.{1}'.format(low['state'], low['fun'])]  # pylint: disable=W0106
      File "/usr/lib/python2.7/dist-packages/salt/utils/lazy.py", line 90, in __getitem__
        raise KeyError(key)
    KeyError: 'git.latest'
```

This wasn't always the case. Salt installs git, then everything that uses `git.latest` requires the `pkg: git`. Looks like when we upgraded our minions/master to `2015.5.3` (last time we updated was from `2014.x.x`), something changed.

```saltstack
{% macro git(git_repo, home, project_name, username=None, rev='master', force_reset=False) %}
{% set ignored = kwargs %}
{{ project_name }}-git:
  git:
    - latest
    {% if force_reset %}
    - force_reset: True
    {% endif %}
    - name: ssh://git@xxx/{{ git_repo }}.git
    - target: {{ home }}/{{ project_name }}
    - identity: /root/.ssh/diffusion_ssh_key
    - rev: {{ rev }}
    - always_fetch: True
    - require:
      - pkg: git
      {% if username %}
      - user: user_{{ username }}
      {% endif %}
{% endmacro %}
```

**Comment by jfindlay** [comment]
@jhgg, thanks for the report.

**Comment by terminalmage** [comment]
A module refresh should be triggered every time a pkg state installs a package, so this shouldn't happen. It would help if we had more information about your minion, so far all we know is that you are using 2015.5.3, and that you're using a jinja macro, but we don't even have a complete idea of how you're calling this macro.

**Comment by terminalmage** [comment]
Also, what OS is the minion running? What version?

**Comment by jhgg** [comment]
Hey, sorry bout that:

# Master 

```
root@salt:~# salt --versions
                  Salt: 2015.5.3
                Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1
               Tornado: Not Installed
 Debian source package: 2015.5.3+ds-1trusty1
root@salt:~# cat /etc/issue
Ubuntu 14.04.3 LTS \n \l
root@salt:~# uname -a
Linux salt.xxx 3.13.0-57-generic #95-Ubuntu SMP Fri Jun 19 09:28:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

# Minion

```
jh@argus:~$ salt-call --versions
                  Salt: 2015.5.3
                Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1
               Tornado: Not Installed
 Debian source package: 2015.5.3+ds-1trusty1
jh@argus:~$ cat /etc/issue
Ubuntu 14.04.3 LTS \n \l
jh@argus:~$ uname -a
Linux argus.prod.xxx 3.13.0-62-generic #102-Ubuntu SMP Tue Aug 11 14:29:36 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by jhgg** [comment]
How the macro is being called:

`component/example/init.sls`
```
{% from "xxx/diffusion/git.jinja" import git %}
{% from "component/example/map.jinja" import config %}
include:
  - packages.util.git

{{ git(**config) }}
```

`packages/util/git/init.sls`
```
git:
  pkg:
    - installed
    - require:
      - pkg: openssh-client

openssh-client:
  pkg:
    - installed
```

`component/example/map.sls`
```
{% set config = {
    'project_name': 'example',
    'username': 'example',
    'home': '/webapps/example',
    'git_repo': 'EX/example',
    'rev': salt['pillar.get']('example:git:rev', 'master')
} %}
```

**Comment by terminalmage** [comment]
Thanks for the additional information, I'll look into this

**Comment by ryanwalder** [comment]
May be similar to my issue (#26241) which looked to be fixed in 2015.5 branch when I lat tested.

**Comment by terminalmage** [comment]
I can confirm this no longer happens. Marking this one closed.
